cmd/gmailhelper.go

```diff
@@ -509,13 +520,14 @@ func (h *GmailHelper) printFilterAndMaybeDiff(filter, newFilter *gm.Filter) {
 	}
 
 	// Make a set of all keys that need to be shown
-	allCriteriaKeys := map[string]byte{}
+	allCriteriaKeys := map[string]string{}
```

Comment:
I think this is supposed to be a set. Though go doesn't have a native set type. Pretty sure I was using byte because it was really small. Maybe making it bool would be more obvious. Making it string -> string is more confusing, I think.

Reply:
The reason I changed it to string->string was to factor out the sortedRange (sortedStringSet) code.  If not, then allCriteriaKeys and actionMap have different types, unless there's an idiomatic way to do this in go?

Reply:
Ah, I didn't notice the other use.

I don't know if there are idiomatic ways of doing this, but there are several ways, though none that I found that were really satisfactory.

I think the one which I like the most, mainly for compile time checking is just to implement (and I think this could just go into some file like util/map.go) is to just implement `StrStrMapKeys` and `StrBoolMapKeys`, which each should be pretty small. Then just giving that to sort.Strings. Or adding something like util/slice.go with 
```
func SortStrSlice(slice []string) []slice {
   sort.Strings(slice)
   return slice
}
```

Then you can use composition (which is, I think idiomatic of golang)
```
for _, s := range util.SortStrSlice(util.StrStrMapKeys(mymap)) {
```

I'll note that your iteration there is two lines longer than it needs to be, since you can write:
```
keys := make(string[], 0, len(mymap))
for k := range mymap {
   keys = append(keys, k)
}
```